examples/rfq/README.md

```diff
@@ -0,0 +1,278 @@
+
+From the root of Linera repository, this can be achieved as follows:
+
+```bash
+export PATH="$PWD/target/debug:$PATH"
+source /dev/stdin <<<"$(linera net helper 2>/dev/null)"
+```
+
+To start the local Linera network:
+
+```bash
+linera_spawn_and_read_wallet_variables linera net up --extra-wallets 1 --testing-prng-seed 37
+```
+
+We use the test-only CLI option `--testing-prng-seed` to make keys deterministic and simplify our
+explanation.
+
+```bash
+export CHAIN_0=e476187f6ddfeb9d588c7b45d3df334d5501d6499b3f9ad5595cae86cce16a65
+export CHAIN_1=1db1936dad0717597a7743a8353c9c0191c14c3a129b258e9743aec2b4f05d03
+export OWNER_0=7136460f0c87ae46f966f898d494c4b40c4ae8c527f4d1c0b1fa0f7cff91d20f
+export OWNER_1=b4f8586041a07323bd4f4ed2d758bf1b9a977eabfd4c00e2f12d08a0899485fd
+```
+
+The `--extra-wallets 1` option creates an additional user chain and wallet - we will use it for the
+user requesting a quote.
+
+Now we have to publish and create the fungible applications.
+
+```bash
+(cd examples/fungible && cargo build --release --target wasm32-unknown-unknown)
+
+APP_ID_0=$(linera --with-wallet 0 project publish-and-create \
+           examples/fungible \
+           --json-argument '{ "accounts": { "User:'$OWNER_0'": "500", "User:'$OWNER_1'": "500" } }' \
+           --json-parameters "{ \"ticker_symbol\": \"FUN1\" }")
+
+APP_ID_1=$(linera --with-wallet 0 project publish-and-create \
+           examples/fungible \
+           --json-argument '{ "accounts": { "User:'$OWNER_0'": "500", "User:'$OWNER_1'": "500" } }' \
+           --json-parameters "{ \"ticker_symbol\": \"FUN2\" }")
+```
+
+Each user is granted 500 tokens of each type.
+
+We also need to publish the bytecode for the Matching Engine application (without instantiating it).
+
+```bash
+ME_BCID=$(linera -w 0 publish-bytecode \
+    examples/target/wasm32-unknown-unknown/release/matching_engine_{contract,service}.wasm)
+```
+
+Lastly, we have to create the RFQ application. We pass in the bytecode ID of the Matching Engine
+bytecode, so that the application can instantiate the Matching Engine on the temporary chains for us.
+
+```bash
+APP_RFQ=$(linera -w 0 --wait-for-outgoing-messages \
+    project publish-and-create examples/rfq \
+    --json-parameters "{\"me_bytecode_id\":\"$ME_BCID\"}" \
+    --required-application-ids $APP_ID_0 $APP_ID_1)
+```
+
+We also need to make sure that both users can access the RFQ application, so we have to request it
+on the second user's chain.
+
+```bash
+linera -w 1 request-application $APP_RFQ
+```
+
+## Using the RFQ Application
+
+First, node services for the both users' wallets have to be started:
+
+```bash
+linera -w 0 service --port 8080 &
+linera -w 1 service --port 8081 &
+```
+
+### Using GraphiQL
+
+Type each of these in the GraphiQL interface and substitute the env variables with their actual
+values that we've defined above.
+
+First, user B has to claim their tokens on their chain, since they were created on user A's chain.
```

Comment:
Would it simplify things if B created the FUN0 app?

It's probably also sufficient if A only has FUN1 and B only has FUN2. (Not sure if that feels less natural as an example.)